AnalyzerTools/src/MCCorrection.C

```diff
@@ -6115,21 +6117,59 @@ void MCCorrection::SetupMCJetTagEff(TString EffFile){
     this_hist->SetDirectory(0);
     cout<<"[MCCorrection::SetupMCJetTagEff] setting "<<hden<<endl;
   }
-  // Numerator histogram setup and divided using "binomial option"
-  for(const auto& obj:*(fmcjet.GetListOfKeys())){
-    TH2F* this_hist=(TH2F*)((TKey*)obj)->ReadObj();
-    TString hnum=this_hist->GetName();
-    if(!hnum.Contains("num")) continue;
-    TString hden="";
-    if(hnum.Contains("_B_")) hden="Jet_"+DataEra+"_eff_B_denom";
-    else if(hnum.Contains("_C_")) hden="Jet_"+DataEra+"_eff_C_denom";
-    else hden="Jet_"+DataEra+"_eff_Light_denom";
-
-    this_hist->Divide(this_hist,map_hist_mcjet[hden],1.,1.,"b");
-    map_hist_mcjet[hnum]=this_hist;
+
+  // Numerator histogram setup and divide using "binomial option"
+  for (const auto& obj : *(fmcjet.GetListOfKeys())) {
+    TH2F* this_hist = (TH2F*)((TKey*)obj)->ReadObj();
+    TString hnum = this_hist->GetName();
+    if (!hnum.Contains("num")) continue;
+
+    TString hden = "";
+    if (hnum.Contains("_B_")) hden = "Jet_" + DataEra + "_eff_B_denom";
+    else if (hnum.Contains("_C_")) hden = "Jet_" + DataEra + "_eff_C_denom";
+    else hden = "Jet_" + DataEra + "_eff_Light_denom";
+
+    // Set any 0 or negative bins to 1
+    for (int x = 1; x <= this_hist->GetNbinsX(); ++x) {
+        for (int y = 1; y <= this_hist->GetNbinsY(); ++y) {
+            double bin_content = this_hist->GetBinContent(x, y);
+            if (bin_content <= 0) {
+                this_hist->SetBinContent(x, y, 1.0);
```

Comment:
Here the idea is if some bins have 0 entires for Jet Efficiency the code changes the hist bin to value 1.